frontend/build.go

```diff
@@ -94,3 +95,21 @@ func BuildWithPlatform(ctx context.Context, client gwclient.Client, f PlatformBu
 	}
 	return rb.Finalize()
 }
+
+// GetBaseImage returns an image that first checks if the client provided the
+// image in the build context matching the image ref.
+//
+// This follows the behavior of of the dockerfile frontend.
+func GetBaseImage(sOpt dalec.SourceOpts, ref string) llb.State {
+	return llb.Scratch().Async(func(ctx context.Context, _ llb.State, c *llb.Constraints) (llb.State, error) {
+		fromClient, err := sOpt.GetContext(ref, dalec.WithConstraint(c))
```

Comment:
Does this allow the user to pass in a base image as a build context?

Reply:
Yes, this is something we already do.